assets/js/phoenix_live_view/rendered.js

```diff
@@ -17,6 +19,90 @@ import {
   isCid,
 } from "./utils"
 
+const VOID_TAGS = [
+  "area",
+  "base",
+  "br",
+  "col",
+  "command",
+  "embed",
+  "hr",
+  "img",
+  "input",
+  "keygen",
+  "link",
+  "meta",
+  "param",
+  "source",
+  "track",
+  "wbr"
+]
+
+export let modifyRoot = (html, attrs, clearInnerHTML) => {
+  let i =0
+  let insideComment = false
+  let insideTag = false
+  let tag
+  let beforeTagBuff = []
+  while(i < html.length){
+    let char = html.charAt(i)
+    if(insideComment){
+      if(char === "-" && html.slice(i, i + 3) === "-->"){
+        insideComment = false
+        beforeTagBuff.push("-->")
+        i += 3
+      } else {
+        beforeTagBuff.push(char)
+        i++
+      }
+    } else if(char === "<" && html.slice(i, i + 4) === "<!--"){
+      insideComment = true
+      beforeTagBuff.push("<!--")
+      i += 4
+    } else if(char === "<"){
+      insideTag = true
+      let iAtOpen = i
+      for(i; i < html.length; i++){
+        if([">", " ", "\n", "\t", "\r"].indexOf(html.charAt(i)) >= 0){ break }
+      }
+      tag = html.slice(iAtOpen + 1, i)
+      break
+    } else if(!insideComment && !insideTag){
+      beforeTagBuff.push(char)
+      i++
+    }
+  }
+  if(!tag){ throw new Error(`malformed html ${html}`) }
+
+  let attrsStr =
+    Object.keys(attrs)
+    .map(attr => attrs[attr] === true ? attr : `${attr}="${attrs[attr]}"`)
+    .join(" ")
+
+  let isVoid = VOID_TAGS.indexOf(tag) >= 0
```

Comment:
For perf, I believe that using a `Set` in combination with `Set.prototype.has` is going to give you better performance than `Array.prototype.indexOf`.

Reply:
If `Set/Set#has` _doesn't_ provide better performance, setting up the list as a an object where the keys are the void_tags and the values are all true will give you constant-time lookups.